### PR TITLE
[Android][iOS] Upgrade @react-native-community/netinfo to 9.3.7

### PR DESCRIPTION
**Pull request by douglowder**
# Why

Closes ENG-7317

# How

```
et uvm -m @react-native-community/netinfo -c 'v9.3.7'
et pods -f
yarn
```

# Test Plan

- [ ] test unversioned expo go ios
- [ ] test unversioned expo go android
- [ ] CI tests pass

# Checklist

- [X] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [x] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.7.2
\ No newline at end of file
+2.7.7
```
Comment (tsapeta) [inline .ruby-version:1]: Is this necessary as part of this PR?

File: packages/html-elements/package.json [hunk 2]
```diff
@@ -38,15 +38,21 @@
     "projects": [
       {
         "preset": "jest-expo/ios",
-        "modulePathIgnorePatterns": ["<rootDir>/babel/"]
+        "modulePathIgnorePatterns": [
```
Comment (tsapeta) [inline packages/html-elements/package.json:41]: It seems to be an unrelated formatting change 😅 I would avoid that if the PR has nothing to do with this particular package.
For example it's then recognized as an unpublished change, but in fact it doesn't have to be published.

**Review by tsapeta — APPROVED** [review]
Looks good, added two nitpicking comments but except that it's ready to go

File: .ruby-version (hunk 1 above)
Comment (douglowder) [reply .ruby-version:1]: This is the only 2.7 version I can get to work on my machine -- I'll revert this

File: packages/html-elements/package.json (hunk 2 above)
Comment (douglowder) [reply packages/html-elements/package.json]: Reverted

File: apps/bare-expo/ios/Podfile.lock [hunk 3]
```diff
@@ -562,7 +562,7 @@ PODS:
     - React-Core
   - react-native-slider (4.2.4):
     - React-Core
-  - react-native-view-shot (3.4.0):
+  - react-native-view-shot (3.5.0):
```
Comment (gabrieldonadel) [inline apps/bare-expo/ios/Podfile.lock:565]: @douglowder shouldn't we update `react-native-netinfo (9.3.7)` here instead?

Comment (douglowder) [reply apps/bare-expo/ios/Podfile.lock:565]: Hmm yes, I will correct that in the next PR
